src/SourceBuild/content/eng/run-binary-tooling.sh

```diff
@@ -0,0 +1,210 @@
+###                              Defaults to online dotnet-public and dotnet-libraries feeds.
+###   --with-sdk                 Use the specified directory as the dotnet SDK.
+###                              Defaults to .dotnet.
+
+set -euo pipefail
+IFS=$'\n\t'
+
+source="${BASH_SOURCE[0]}"
+REPO_ROOT="$( cd -P "$( dirname "$0" )/../" && pwd )"
+BINARY_TOOL="$REPO_ROOT/eng/tools/BinaryToolKit"
+
+function print_help () {
+    sed -n '/^### /,/^$/p' "$source" | cut -b 5-
+}
+
+defaultDotnetSdk="$REPO_ROOT/.dotnet"
+
+# Set default values
+baseline=''
+mode=''
+logLevel='Debug'
+propsDir=''
+packagesDir=''
+dotnetSdk=$defaultDotnetSdk
+
+positional_args=()
+while :; do
+  if [ $# -le 0 ]; then
+    break
+  fi
+  lowerI="$(echo "$1" | awk '{print tolower($0)}')"
+  case $lowerI in
+    "-?"|-h|--help)
+      print_help
+      exit 0
+      ;;
+    --clean)
+      mode="clean"
+      if [ -z "$baseline" ]; then
+        baseline="$REPO_ROOT/src/installer/src/VirtualMonoRepo/allowed-sb-binaries.txt"
+      fi
+      ;;
+    --validate)
+      mode="validate"
+      if [ -z "$baseline" ]; then
+        baseline="$REPO_ROOT/src/installer/src/VirtualMonoRepo/allowed-vmr-binaries.txt"
+      fi
+      ;;
+    --baseline)
+      baseline=$2
+      if [ ! -f "$baseline" ]; then
+        echo "ERROR: The specified baseline file does not exist."
+        exit 1
+      fi
+      shift
+      ;;
+    --log-level)
+      logLevel=$2
+      shift
+      ;;
+    --with-packages)
+        packagesDir=$2
+        if [ ! -d "$packagesDir" ]; then
+            echo "ERROR: The specified packages directory does not exist."
+            exit 1
+        elif [ ! -f "$packagesDir/PackageVersions.props" ]; then
+            echo "ERROR: The specified packages directory does not contain PackageVersions.props."
+            exit 1
+        fi
+        shift
+        ;;
+    --with-sdk)
+        dotnetSdk=$2
+        if [ ! -d "$dotnetSdk" ]; then
+            echo "Custom SDK directory '$dotnetSdk' does not exist"
+            exit 1
+        fi
+        if [ ! -x "$dotnetSdk/dotnet" ]; then
+            echo "Custom SDK '$dotnetSdk/dotnet' does not exist or is not executable"
+            exit 1
+        fi
+        shift
+        ;;
+    *)
+      positional_args+=("$1")
+      ;;
+  esac
+
+  shift
+done
+
+function ParseBinaryArgs
+{
+    if [ -z "$mode" ]; then
+        echo "ERROR: --clean or --validate must be specified."
+        print_help
+        exit 1
+    fi
+
+    # Check dotnet sdk
+    if [ "$dotnetSdk" == "$defaultDotnetSdk" ]; then
+        if [ ! -d "$dotnetSdk" ]; then
+            . "$REPO_ROOT/eng/common/tools.sh"
+            InitializeDotNetCli true
+        fi
+        else if [ ! -x "$dotnetSdk/dotnet" ]; then
+            echo "'$dotnetSdk/dotnet' does not exist or is not executable"
+            exit 1
+        fi
+    fi
+
+    # Check the packages directory
+    if [ -z "$packagesDir" ]; then
```

Comment:
I don't think we should be dynamically retrieving a version.  It should be statically specified in the versions.props file.  SB will override it when --with-packages is specified.